src/Compilers/Core/Portable/Operations/Operation.cs

```diff
@@ -92,6 +90,8 @@ public string Language
 
         public abstract IEnumerable<IOperation> Children { get; }
 
+        public SemanticModel SemanticModel { get; }
```

Comment:
I think this is MemberSemanticModel not the TreeSemanticModel people would usually get from Compilation.GetSemanticModel.

I might be wrong. but you should check. and I think we should always return TreeSemanticModel since that's the only one we expose. Basically SemanticModel.Parent

Reply:
I think we decided to expose the semantic model that was used to create the operation, which is the MemberSemanticModel. Tagging @AlekseyTs @gafter @tmat @DustinCampbell @333fred @jinujoseph to confirm.

Reply:
do we ever exposed those before? and one of problem is that treeSemanticModel.GetOperation works for every node from same treeSemanticMode.SyntaxTree but MemberSemanticModel.GetOperation only works for ones that belong to that member I believe?

question will be how user know?

Reply:
> I will make this change once I get a confirmation from the rest that we want to expose the syntax tree model.

Yes, we want to expose an instance that a client can actually get from compilation, not any of the instances used internally by compilers. Basically, one should get an instance that client called GetOperation API on. 


---
In reply to: [195875720](https://github.com/dotnet/roslyn/pull/27922#discussion_r195875720) [](ancestors = 195875720)

Reply:
Also, it is possible that for proper working of the lazy Parent calculation, we actually need to use the internal SemanticModel, so we might need to store one SemanticModel in the node and return something different through the interface. For example, the clonner should probably keep working with internal SemanticModel 

---
In reply to: [196165280](https://github.com/dotnet/roslyn/pull/27922#discussion_r196165280) [](ancestors = 196165280,195875720)